src/sqlfluff/core/dialects/ansi_keywords.py

```diff
@@ -19,6 +19,7 @@
 PARTITION
 ORDER
 ROWS
+SET
```

Comment:
Without this change, the parser couldn't handle `UPDATE` statements without aliases. This helps the parser determine the end of the table reference and the start of the "set clause" that updates the data.